refactor(ast/ast_builder)!: shorter allocator utility method names.

rzvxa opened the pull request:
This PR serves two purposes, First off it would lower the amount of characters we have to type in for a simple operation such as wrapping an expression in a vector. Secondly, it would follow the generated names more closely since nowhere else in the builder we do have `new_xxx`, We always say `xxx` since a builder always constructs something.

```
new_vec -> vec
new_vec_single -> vec1*
new_vec_from_iter -> vec_from_iter
new_vec_with_capacity -> vec_with_capacity
new_str -> str
new_atom -> atom
```

`*` This one is the main motivation behind this PR, It saves 10 characters!

rzvxa commented:
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/4122?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#4122** <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/4122?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a> 👈
* **#4117** <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/4117?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a>
* **#4116** <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/4116?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a>
* **#3890** <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/3890?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a>
* **#4098** <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/4098?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a>
* **#4090** <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/4090?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a>
* **#4088** <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/4088?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a>
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @rzvxa and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="11px" height="11px"/> <b>Graphite</b></a>
  <!-- Current dependencies on/for this PR: -->

overlookmotel commented:
I agree with the logic of this change. I'd suggest `vec_from_iter` rather than `vec_from`. That might be an abbreviation too far (from what?) - and also it's not much used.

rzvxa commented:
> I agree with the logic of this change. I'd suggest `vec_from_iter` rather than `vec_from`. That might be an abbreviation too far (from what?) - and also it's not much used.

Sorry, my summary didn't cover all of the changes, I've changed the signature of our former `new_vec_from_iter` to this:

```rust
    #[inline]
    pub fn vec_from<T, I: IntoIterator<Item = T>>(self, iter: I) -> Vec<'a, T> {
        Vec::from_iter_in(iter, self.allocator)
    }
```


So it's no longer just "from_iter", It accepts anything that can be iterated even something like `Some(T)`. That's why I've made the name more generic as well.

overlookmotel commented:
I still prefer `vec_from_iter` as it [matches std's naming convention](https://doc.rust-lang.org/std/vec/struct.Vec.html#method.from_iter). We could conceivably want to add e.g. `vec_from_slice` later on. But I don't feel so strongly about it.

rzvxa commented:
> I agree with the logic of this change. I'd suggest `vec_from_iter` rather than `vec_from`. That might be an abbreviation too far (from what?) - and also it's not much used.

At first, I thought it was accepting iterators before so I sent a false comment. I removed it to write an accurate answer instead.

I chose this name because it doesn't accept `Iterator`, It accepts `IntoIterator` which can be a variety of things such as `Some(T)` so that's why I went with this generic name. Let me know if you don't follow my reasoning or if you have a relatively strong objection, I'm flexible for changing it.

rzvxa commented:
> I still prefer `vec_from_iter` as it [matches std's naming convention](https://doc.rust-lang.org/std/vec/struct.Vec.html#method.from_iter) (`Vec::from_iter` also accepts anything which can be iterated). We could conceivably want to add e.g. `vec_from_slice` later on. But I don't feel so strongly about it.

I agree with your reasoning, Just pushed a new commit addressing this.

rzvxa committed:
refactor(ast/ast_builder)!: shorter allocator utility method names. (#4122)

This PR serves two purposes, First off it would lower the amount of characters we have to type in for a simple operation such as wrapping an expression in a vector. Secondly, it would follow the generated names more closely since nowhere else in the builder we do have `new_xxx`, We always say `xxx` since a builder always constructs something.

```
new_vec -> vec
new_vec_single -> vec1*
new_vec_from_iter -> vec_from_iter
new_vec_with_capacity -> vec_with_capacity
new_str -> str
new_atom -> atom
```

`*` This one is the main motivation behind this PR, It saves 10 characters!
